openpype/plugins/load/add_site.py

```diff
@@ -12,21 +22,61 @@ class AddSyncSite(load.LoaderPlugin):
     icon = "download"
     color = "#999999"
 
+    _sync_server = None
+
+    @property
+    def sync_server(self):
+        if not self._sync_server:
+            manager = ModulesManager()
+            self._sync_server = manager.modules_by_name["sync_server"]
+
+        return self._sync_server
+
     def load(self, context, name=None, namespace=None, data=None):
         self.log.info("Adding {} to representation: {}".format(
             data["site_name"], data["_id"]))
-        self.add_site_to_representation(data["project_name"],
-                                        data["_id"],
-                                        data["site_name"])
+        family = context["representation"]["context"]["family"]
+        project_name = data["project_name"]
+        repre_id = data["_id"]
+
+        add_ids = [repre_id]
+        if family == "workfile":
+            links = get_linked_ids_for_representations(project_name,
+                                                       add_ids,
+                                                       link_type="reference")
+            add_ids.extend(links)
+
+        add_ids = set(add_ids)
+        self.log.info("Add to repre_ids {}".format(add_ids))
+        is_main = True
+        for add_repre_id in add_ids:
+            self.add_site_to_representation(project_name,
+                                            add_repre_id,
+                                            data["site_name"],
+                                            is_main)
+            is_main = False
```

Comment:
Are you setting it only on the first entry because you're expecting the first entry to be the initial `repre_id`? If so, then your `is_main` logic is wrong because you're turning it into a `set` to ensure unique entries which will mangle up the order.

Unlike `dict` even in Python 3 sets are unsorted. See also [here](https://stackoverflow.com/a/45589593/1838864).

Here's an alternative. **Pseudocode!**
```suggestion
        repre_id = data["_id"]
        self.add_site_to_representation(project_name,
                                        repre_id,
                                        data["site_name"],
                                        is_main=True)

        if family == "workfile":
            links = get_linked_ids_for_representations(project_name,
                                                       add_ids,
                                                       link_type="reference")
            for link_repre_id in links:
                self.add_site_to_representation(project_name,
                                                link_repre_id,
                                                data["site_name"],
                                                is_main=False)
```